lib/_http_agent.js

```diff
@@ -115,6 +115,9 @@ Agent.prototype.addRequest = function(req, options) {
     };
   }
 
+  options = util._extend({}, options);
```

Comment:
I think it was done this way in order to remain consistent with the rest of the file. If we're refactoring this to an `Object.assign` we might as well refactor the one in `createSocket` to use `Object.assign`